### ISSUE TITLE
[BUG] st_read fails to read Excel file from an http URL

### ISSUE THREAD

**Issue by tboddyspargo** [duckdb/duckdb_spatial#181]
In duckdb v0.9.2, the following code fails (where it succeeded in 0.9.1):

```
import duckdb

conn = duckdb.connect(database=':memory:')
conn.execute('SET s3_endpoint="localhost:8333";')
conn.execute('SET s3_url_style="path";')
conn.execute('SET s3_use_ssl=false;')
conn.execute('INSTALL spatial; LOAD spatial;')
r = conn.execute(
    "SELECT * FROM st_read('http://localhost:8333/appdata/global/sba_sample.xlsx', layer='sba_sample')"
)

print(r.fetchdf())
```

In this case, `http://localhost:8333` is a local seaweedfs container compliant with the S3 API (and `sba_sample.xlsx` exists at that path).

With this exception:

```
Traceback (most recent call last):
  File "tmp/excel_example.py", line 7, in <module>
    r = conn.execute(
duckdb.duckdb.IOException: IO Error: Unknown file type
```

**Comment by tboddyspargo** [comment]
I'll note that this was working in duckdb `0.9.1` and it has been our work-around due to the other failure mentioned in #182. 

Thanks in advance for your help!

**Comment by tboddyspargo** [comment]
I suspect this may have something to do with #168, given that the error message matches.

**Comment by Maxxen** [comment]
Hi! I think this has been fixed in https://github.com/duckdb/duckdb_spatial/pull/187. 

Note: for XLSX in particular using DuckDB's filesystem with httpfs has a slower initial load time than `/vsis3/` or `/vsicurl/`. Im aware of the issue and will look into improving it more in the future.

In a couple of hours (when the CI finishes) you should be able to install the extension with this fix for 0.9.2 by running:

```sql
FORCE INSTALL spatial FROM 'http://nightly-extensions.duckdb.org';
```

**Comment by tboddyspargo** [comment]
Thanks @Maxxen - Can fix be made available in the default 0.9.2 `spatial` extension or in a 0.9.3 fix release?

**Comment by Maxxen** [comment]
We may eventually replace the default 0.9.2 install version with this fixed version, but I can't give a timeline. If more issues occur it may be delayed further.
